src/theory/quantifiers/query_generator.cpp

```diff
@@ -135,9 +141,14 @@ bool QueryGenerator::addTerm(Node n, std::ostream& out)
     if (qsi.size() > 1)
     {
       // take two random queries
-      std::shuffle(qsi.begin(), qsi.end(), Random::getRandom());
-      Node qy = nm->mkNode(AND, qsi[0], qsi[1]);
-      checkQuery(qy, i);
+      size_t rindex = Random::getRandom().pick(0, qsi.size() - 1);
+      size_t rindex2 = Random::getRandom().pick(0, qsi.size() - 1);
+      if (rindex == rindex2)
+      {
+        rindex2 = rindex + 1 == qsi.size() ? 0 : rindex + 1;
```

Comment:
I guess we don't care that the distribution of pairs is not uniform anymore, right? Can we maybe add a comment about that?

Reply:
It's not a big deal, I'd prefer this than a while loop to avoid collisions.

Reply:
Actually its simple to make it uniform, I've updated.